azurelinuxagent/common/protocol/extensions_goal_state_from_vm_settings.py

```diff
@@ -325,7 +335,52 @@ def _parse_extensions(self, vm_settings):
                     extension.manifest_uris.extend(additional_locations)
 
-                # TODO: Add settings
+                #
+                # Settings
+                #
+                settings_list = extension_gs.get('settings')
+                if settings_list is not None:
+                    if not isinstance(settings_list, list):
+                        raise Exception("'settings' should be an array (extension: {0})".format(extension.name))
+                    if not extension.supports_multi_config and len(settings_list) > 1:
+                        raise Exception("Single-config extension includes multiple settings (extension: {0})".format(extension.name))
+
+                    for s in settings_list:
+                        settings = ExtensionSettings()
+                        public_settings = s.get('publicSettings')
+                        settings.publicSettings = {} if public_settings is None else json.loads(public_settings)
+                        # Note that protectedSettings and protectedSettingsCertThumbprint can be None (which would be serialized to the extension's settings file as null)
+                        settings.protectedSettings = s.get('protectedSettings')
+                        thumbprint = s.get('protectedSettingsCertThumbprint')
+                        if thumbprint is None and settings.protectedSettings is not None:
+                            raise Exception("The certificate thumbprint for protected settings is missing (extension: {0})".format(extension.name))
+                        settings.certificateThumbprint = thumbprint
+
+                        # in multi-config each settings have their own name, sequence number and state
+                        if extension.supports_multi_config:
+                            settings.name = s['extensionName']
+                            settings.sequenceNumber = s['seqNo']
+                            settings.state = s['extensionState']
+                        else:
+                            settings.name = extension.name
+                            settings.sequenceNumber = extension_gs['settingsSeqNo']
+                            settings.state = extension.state
+                        extension.settings.append(settings)
+
+                #
+                # Dependency level
+                #
+                depends_on = extension_gs.get("dependsOn")
+                if depends_on is not None:
+                    if not isinstance(depends_on, list):
+                        raise Exception('dependsOn should be an array')
+                    if extension.supports_multi_config:
+                        # TODO: Parse dependencies for multi-config extensions
+                        pass
+                    else:
+                        if len(depends_on) != 1:
+                            raise Exception('dependsOn should be an array with exactly one item for single-config extensions')
```

Comment:
I dont get this condition. There can be extensions with multiple dependencies, eg: `A -> B, C, D`. That would fail this condition check right?

Reply:
let me double check that on an actual run... i thought multiple dependencies would be expressed in the DependsOnExtension array

```
            "dependsOn": [
                {
                    "DependsOnExtension": [
                        {
                            "handler": "Microsoft.Azure.Security.Monitoring.AzureSecurityLinuxAgent"
                        }
                    ],
                    "dependencyLevel": 1
                }
            ]

```

Reply:
yes, that seems to be the case (they are in DependsOnExtension)

            "dependsOn": [
                {
                    "DependsOnExtension": [
                        {
                            "handler": "Microsoft.CPlat.Core.RunCommandLinux"
                        },
                        {
                            "handler": "Microsoft.Azure.Extensions.Edp.GATestExtGo"
                        }
                    ],
                    "dependencyLevel": 2
                }
            ]